lib/stat_tracker.rb

```diff
@@ -29,44 +29,29 @@ def lowest_total_score
   def biggest_blowout
     @game_collection.games.map {|game| (game.away_goals - game.home_goals).abs}.max
   end
-  # This only requires game information.
-  # It should probably move to game collection eventually.
-  def count_of_games_by_season
-    seasons = @game_collection.games.map{|game| game.season }.uniq  #season needs helper method
-    seasons.reduce({}) do |games_by_season, season|
-      games_per_season = @game_collection.games.find_all do |game|  #games_per_season needs helper method
-         season == game.season
-      end.length
-    games_by_season[season] = games_per_season
-    games_by_season
-    end
+
+                                                                    # This only requires game information.
+                                                                    # It should probably move to game collection eventually.
+  def count_of_games_by_season                                      #refactored by Ryan 2.22.20
+    games_by_season = @game_collection.all.group_by{|game| game.season}         #games_by_season 1st occurance
+    games_by_season.transform_values!{|games| games.length}
   end
 
-  # Can and should implement daniels helper method for total_goals_per_game
   # This only requires game information.
   # It should probably move to game collection eventually.
   def average_goals_per_game
-    total_goals_per_game = @game_collection.games.map do |game|
-      game.home_goals + game.away_goals
-    end
-    average = total_goals_per_game.sum / total_goals_per_game.length.to_f  # create module with average method??
-    average.round(2)
+    all_goals = @game_collection.total_goals_per_game
```

Comment:
I wonder if we could save all_goals as a method in the GameCollection class. This would help remove a line of code from this method in StatTracker